PRChecker/Shared/ViewModels/FilterViewModel.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  FilterViewModel.swift
+//  PRChecker
+//
+//  Created by Bruce Evans on 2021/11/07.
+//
+
+import Foundation
+
+class FilterViewModel: ObservableObject {
+    
+    static let statusFilter: FilterSection = {
+        FilterSection(
+            name: "Status",
+            filters: [
+                Filter(name: "Open") { $0.state == .open },
+                Filter(name: "Merged") { $0.state == .merged },
+            ]
+        )
+    }()
+    
+    static let reviewStatusFilter: FilterSection = {
+        FilterSection(
+            name: "Review Status",
+            filters: [
+                Filter(name: "Waiting") { $0.viewerStatus == .waiting },
+                Filter(name: "Blocked") { $0.viewerStatus == .blocked },
+                Filter(name: "Commented") { $0.viewerStatus == .commented },
+                Filter(name: "Approved") { $0.viewerStatus == .approved },
+            ]
+        )
+    }()
+    
+    lazy var sections: [FilterSection] = [
+        Self.statusFilter,
+        Self.reviewStatusFilter,
+        // TODO: Label Filter
```

Comment:
Hope we can filter by repository name and read/unread status too in the future

Reply:
Yep. I was able to get the labels working, and the process is the same, so it should be possible to filter on anything in the PullRequestCell

Reply:
Nice! Let me merge this PR first.